File: 백준/Bronze/2738. 행렬 덧셈/행렬 덧셈.py
import sys
input = sys.stdin.readline

def add_ab(alist: list[int], blist: list[int]):
    for row_a, row_b in zip(alist, blist, strict=True):
        for a, b in zip(row_a, row_b, strict=True):
            print(a + b, end=" ")
        print()

def solve():
    line = input().split()
    if not line: return
    n, m = map(int, line)
    
    alist = []
    for _ in range(n):
        line = input().split()
        if not line: return
        alist.append(list(map(int, line)))
        
    blist = []    
    for _ in range(n):
        line = input().split()
        if not line: return
        blist.append(list(map(int, line)))
        
    add_ab(alist, blist)
    
if __name__ == "__main__":
    solve()

File: 백준/Bronze/2738. 행렬 덧셈/행렬 덧셈.cc
#include <iostream>
#include <vector>
using namespace std;

void add_ab(const vector<vector<int>>& alist, const vector<vector<int>>& blist, int n, int m) {
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
			cout << alist[i][j] + blist[i][j] << (j == m - 1 ? "" : " ");
		}
		cout << "\n";
	}
}

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n = 0, m = 0;
	if (!(cin >> n >> m)) return 0;

	vector<vector<int>> alist(n, vector<int>(m, 0));
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
			if (!(cin >> alist[i][j])) return 0;
		}
	}
	vector<vector<int>> blist(n, vector<int>(m, 0));
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
			if (!(cin >> blist[i][j])) return 0;
		}
	}

	add_ab(alist, blist, n, m);

	return 0;
}